components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -212,18 +257,36 @@ def run_clp(
 
     # Compute the total amount of data compressed
     last_archive_stats = None
-    total_uncompressed_size = 0
-    total_compressed_size = 0
-    while True:
+    last_line_decoded = False
+    worker_output = {
+        "total_uncompressed_size": 0,
+        "total_compressed_size": 0,
+    }
+
+    while not last_line_decoded:
         line = proc.stdout.readline()
-        if not line:
-            break
-        stats = json.loads(line.decode("ascii"))
-        if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
+        stats: Optional[Dict[str, Any]] = None
+        if line:
+            stats = json.loads(line.decode("ascii"))
+        else:
+            last_line_decoded = True
+
+        if last_archive_stats is not None and (
+            None is stats or stats["id"] != last_archive_stats["id"]
+        ):
+            if enable_s3_write:
+                result = upload_single_file_archive_to_s3(
+                    last_archive_stats, archive_output_dir, s3_config
+                )
+                if not result.success:
+                    worker_output["error_message"] = result.error
+                    s3_write_failed = True
+                    break
```

Comment:
One limitation we probably want to address is if:
* the compression task is particularly big (e.g., a user requests compression of a huge file that we can't split amongst tasks), and
* uploading archives fails, and
* the user's disk is not very large,
then the disk could get filled up with the archives.

One solution is to simply delete the archives even though they can't be uploaded. Do you see any significant downsides to that?

Reply:
should be resolved